openmetadata-ui/src/main/resources/ui/src/pages/MetricsPage/AddMetricPage/AddMetricPage.tsx

```diff
@@ -289,6 +283,24 @@ const AddMetricPage = () => {
                   onFinish={handleSubmit}
                   onFocus={handleFieldFocus}>
                   {generateFormFields(formFields)}
+                  <Form.Item
+                    label={t('label.unit-of-measurement')}
+                    name="unitOfMeasurement">
+                    <CustomUnitSelect
+                      customValue={form.getFieldValue(
+                        'customUnitOfMeasurement'
+                      )}
+                      dataTestId="unitOfMeasurement"
+                      placeholder={t('label.select-field', {
+                        field: t('label.unit-of-measurement'),
+                      })}
+                      value={form.getFieldValue('unitOfMeasurement')}
+                      onChange={handleUnitOfMeasurementChange}
+                    />
+                  </Form.Item>
+                  <Form.Item hidden name="customUnitOfMeasurement">
+                    <input type="hidden" />
+                  </Form.Item>
```

Comment:
Why it's hidden?

Reply:
We have two fields "unitOfMeasurement" and "customUnitOfMeasurement", when the first has value "OTHER" then only we want this "customUnitOfMeasurement" to be set, handling this through hidden field.